subprojects/core-api/src/main/java/org/gradle/api/artifacts/component/BuildIdentifier.java

```diff
@@ -20,13 +20,25 @@
  * Identifies a Gradle build. The identifier is unique within a Gradle invocation, so for example, each included build will have a different identifier.
  */
 public interface BuildIdentifier {
+
+    /**
+     * Absolute build path of the build within the Gradle invocation.
+     *
+     * @since 8.2
+     */
+    String getBuildPath();
+
     /**
      * The name of the build.
+     *
+     * @deprecated Use {@link #getBuildPath()} instead.
      */
+    @Deprecated
     String getName();
 
     /**
      * Is this build the one that's currently executing?
      */
+    @Deprecated
```

Comment:
❌ Please add a `@deprecated` Javadoc what folks should do instead.

Reply:
I thought about this, but didn't come up with a reasonable replacement explanation. It is not like the build path method still provides this information.
Do you think it is enough to add the same deprecation message here like for the `getName()`?
```
@deprecated Use {@link #getBuildPath()} instead.
```

Reply:
I think you can use the build path to infer whether or not you are in the same build.

Reply:
You are right. Added the deprecation message